lib/decorators.rb

```diff
@@ -3,7 +3,19 @@ module Decorators
 
   class << self
     def load!(cache_classes)
+      decorators_with_argument_errors = []
       decorators.each do |decorator|
+        begin
+          if cache_classes
+            require decorator
+          else
+            load decorator
+          end
```

Comment:
this block could be moved into a `load_decorator` private method:

```ruby
def load_decorator(decorator)
  if cache_classes
    require decorator
  else
    load decorator
  end
end
```

Usage something like:

```ruby
decorators_with_argument_errors = decorators.reduce([]) do |errors, decorator|
  begin  
	load_decorator(decorator) 
  rescue ArgumentError
    errors << decorator
  end
end

decorators_with_argument_errors.each(&method(:load_decorator))
```

That way we avoid the duplication